service/src/main/java/com/derpgroup/quip/manager/QuipManager.java

```diff
@@ -16,39 +19,94 @@
 import com.derpgroup.quip.QuipMetadata;
 
 public class QuipManager extends AbstractManager {
+  private final Logger LOG = LoggerFactory.getLogger(QuipManager.class);
   
   static{
     ConversationHistoryUtils.getMapper().registerModule(new MixInModule());
   }
   
   private static final String[] metaRequestSubjects = new String[]{"ANOTHER"};
+  protected static final int MAXIMUM_QUIP_HISTORY_SIZE = 10;
+  protected static final double MAXIMUM_QUIP_HISTORY_PERCENT = .5;
   
   public QuipManager(){
     super();
   }
   
-  protected void doInsultRequest(Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
+  protected int determineMaxQuipHistorySize(int sizeOfQuipGroup){
+    return Math.min(MAXIMUM_QUIP_HISTORY_SIZE, (int) (sizeOfQuipGroup*MAXIMUM_QUIP_HISTORY_PERCENT));
+  }
+  
+  protected Insults doInsultRequest(Map<String, String> messageMap, SsmlDocumentBuilder builder, QuipMetadata metadata) {
+    Deque<String> insultsUsed = metadata.getInsultsUsed();
```

Comment:
Some portion of the next 11 lines could probably be extracted into its own method or methods and shared across all (currently) 4 of these methods.